common/lib/xmodule/xmodule/course_module.py

```diff
@@ -909,6 +917,8 @@ def __init__(self, *args, **kwargs):
         if self.discussion_topics == {}:
             self.discussion_topics = {_('General'): {'id': self.location.html_id()}}
 
+        self.set_default_user_partition()
```

Comment:
@cpennington @ormsbee What are your thoughts about persisting a change to course_module via the init method? @nasthagiri and I have been talking about it, and we are not sure if there is a better solution.

The background here is that we are introducing a new UserPartition type that is based on enrollment track (so we can ultimately get rid of the cohorting/enrollment track beta feature and instead use a UserPartition that dynamically gets the learner's enrollment mode). We want this partition to be available for all courses.

With the code as-is, if the course_module is loaded from LMS and no enrollment track partition exists, we will create one. This means that the "published" version of the course will end up persisting user_partitions through the save method in caching_descriptor_system's "xblock_from_json" method.

There are other fields already being set in this way (discussion_topics, grading policy, etc.), so I don't think this is a new issue. But is there a better way to do this?